src/app.js

```diff
@@ -10,15 +10,24 @@ const {NotFound} = require('http-errors');
 
 const production = process.env.NODE_ENV === "production";
 
-const charon = require("./endpoints/charon");
-const users = require("./endpoints/users");
-const {auspiceAssets, gatsbyAssets, sendAuspiceEntrypoint, sendGatsbyEntrypoint, sendGatsbyPage, sendGatsby404} = require("./endpoints/static");
-const {setSource, setGroupSource, setDataset, setNarrative, canonicalizeDataset, ifDatasetExists, ifNarrativeExists} = require("./endpoints/sources");
 const authn = require("./authn");
+const endpoints = require("./endpoints");
 const redirects = require("./redirects");
 
+const {
+  setSource,
+  setGroupSource,
+  setDataset,
+  setNarrative,
+  canonicalizeDataset,
+  getDataset,
+  getNarrative,
+} = endpoints.sources;
```

Comment:
It's merely preference. I noted why in the message of the introducing commit (3d65697e4dff56e6a42b50a87ab311da044a377a):

> Source-related endpoints are still destructured into individual
variables because they're used in so many routes and repeating
"endpoints.sources." all over the place would be harder to read.

Reply:
Hmm, I understand not wanting _too much_ verbosity, but since we have multiple `getDataset` handlers I think it would really help to convey _which_ endpoint category¹ they come from, especially helpful when doing "find getDataset" across the codebase, for example.

We could remove some of the verbosity like so:

```js
const {charon, sources} = endpoints;

app.routeAsync("/charon/getDataset")
  .getAsync(charon.getDataset);

app.routeAsync(coreBuildRoutes)
  .all(sources.setDataset(req => req.path), sources.canonicalizeDataset(path => `/${path}`))
  .getAsync(sources.getDataset)
;

```
---

¹ is there a better word for this? I'm referring here to the organisation of the files in `endpoints` which map nicely to different APIs we have.

Reply:
Ok. I can do `const {charon, sources} = endpoints`. Another alternative is splitting up the dataset and narrative-specific bits into separate modules and leaving the shared bits in `src/endpoints/sources.js`. It'd look something like this: https://gist.github.com/tsibley/af7ef26a07781e00e0e56cf1bd8c246d

> is there a better word for this? I'm referring here to the organisation of the files in `endpoints` which map nicely to different APIs we have.

"endpoint module" or "endpoint collection", perhaps?

Reply:
@jameshadfield Any preference on which approach you'd like me to take?